### PR TITLE
FIX Tickets #120 #122 #130

### PR DESCRIPTION
**Pull request by gyfantidis**
- #120:
  - #122:
  - #130

**Description**:

- Fix the double split bug
- Implement the task, delete embeddings, embeddingGroup and sections when a document has updated
- update the save document file path

### Related Tickets: #120 #122 #120

### CODE REVIEW

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/EmbeddingService.java [hunk 1]
```diff
@@ -219,6 +215,30 @@ public Message createMessage(Message message) {
         return message;
     }
 
+    public void deleteEmbeddings(List<Embedding> embeddings) throws GendoxException {
+        for (Embedding embedding : embeddings) {
+            this.deleteEmbedding(embedding.getId());
+        }
```
Comment (sekasx) [inline gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/EmbeddingService.java:221]: embeddingRepository.deleteAll(embeddings);

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/EmbeddingService.java [hunk 2]
```diff
@@ -219,6 +215,30 @@ public Message createMessage(Message message) {
...
+    }
+
+    public void deleteEmbeddingGroupsBySection(UUID sectionId) throws GendoxException {
+        List<EmbeddingGroup> embeddingGroups = embeddingGroupRepository.findBySectionId(sectionId);
+
+        for (EmbeddingGroup embeddingGroup : embeddingGroups) {
+            UUID embeddingId = embeddingGroup.getEmbeddingId();
+            deleteEmbeddingGroup(embeddingGroup.getId());
+            deleteEmbedding(embeddingId);
+        }
```
Comment (sekasx) [inline gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/EmbeddingService.java:239]: Change to deleteAll

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/spring/batch/jobs/splitter/steps/DocumentSplitterProcessor.java [hunk 3]
```diff
@@ -43,26 +43,9 @@ public DocumentSectionDTO process(DocumentInstance item) throws Exception {
...
-
-
-        } catch (Exception e) {
-            logger.warn("Error {} split document to sections {}. Skipping...", e.getMessage(), item.getId());
-            return null;
-        }
-
-
-        return new DocumentSectionDTO(item, contentSections, agent);
+        return new DocumentSectionDTO(item, fileContent, agent);
```
Comment (sekasx) [inline gendox-core-api/src/main/java/dev/ctrlspace/gendox/spring/batch/jobs/splitter/steps/DocumentSplitterProcessor.java:49]: keep split here. 
save sections in the writer

**Review by sekasx — CHANGES_REQUESTED** [review]

File: gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/UploadService.java [hunk 4]
```diff
@@ -45,26 +48,31 @@ public DocumentInstance uploadFile(MultipartFile file, UUID organizationId, UUID
         DocumentInstance instance =
                 documentService.getDocumentByFileName(projectId, organizationId, fileName);
 
+        // find agent's userID
+        UUID userId = projectAgentService.getAgentByProjectId(projectId).getUserId();
```
Comment (myrtp) [inline gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/services/UploadService.java:52]: the userID can be extracted directly from the project by by getting the projectAgent and then the userID. Therefore there is no need to autowire the projectAgentService

**Pull request merged**